### DISCUSSION TITLE
Checking list of operators used in ONNX model

### DISCUSSION THREAD

**Discussion by harishch4** [onnx/onnx#3554] [Q&A] [answered]
Hello Everyone,

Is there a way to get the list of operators used by an ONNX model?

Thanks in advance.

**Comment by askhade** [comment, marked as answer]
You can parse the graphproto and look at the op_type of each NodeProto
Something like this:
auto* g = model_proto.mutable_graph()
for (const auto& n : g->node()) {
    std::cout<< n.op_type() << std::endl;
}

**Reply by hylandk-movidius** [reply]
Hi -

Is there a way to analyze an ONNX model and look for operator types using the PythonAPI?

thanks,
Kevin

**Reply by harishch4** [reply]
I wrote a simple python code to read the ops in the model. 

```
import onnx
model = onnx.load('mobilenetv2-7.onnx')
ops = set()
for node in model.graph.node:
  ops.add(node.op_type)
print(ops)
```

**Reply by hylandk-movidius** [reply]
You beauty - that did it. Thank you

I've tried to find documentation on the attributes/members of the model class when loaded (to do more analysis). Do you know if they are documented anywhere? Like, how did you know there was a member called graph (and a member called node in graph)? 

![image](https://user-images.githubusercontent.com/35958291/150994786-2921d441-f716-4df6-9e2f-7de6ac683140.png)

**Reply by harishch4** [reply]
You can find the relevant information in the below links.

https://github.com/onnx/onnx/blob/main/docs/IR.md
https://github.com/onnx/onnx/blob/main/onnx/onnx.proto3

**Reply by hylandk-movidius** [reply]
Thank you so much. I missed that folder completely. This is perfect. Really appreciate the dig out. K
